llvm/lib/CASObjectFormats/FlatV1.cpp

```diff
@@ -275,17 +275,15 @@ Expected<SectionRef> SectionRef::create(CompileUnitBuilder &CUB,
   if (!B)
     return B.takeError();
 
-  B->Data.push_back(
-      uint8_t(data::encodeProtectionFlags(S.getProtectionFlags())));
+  B->Data.push_back(uint8_t(S.getMemProt()));
 
   B->Data.append(S.getName());
 
   return get(B->build());
 }
 
 Error SectionRef::materialize(LinkGraphBuilder &LGB, unsigned Idx) const {
-  auto Flags =
-      decodeProtectionFlags((data::SectionProtectionFlags)getData()[0]);
+  auto Flags = (jitlink::MemProt)getData()[0];
```

Comment:
Same comment here. Please update `decodeProtectionFlags` rather than assuming the `jitlink` enum has a stable numbering (which it does not).